packages/cache/src/internal/downloadUtils.ts

```diff
@@ -258,21 +260,41 @@ export async function downloadCacheStorageSDK(
         )
 
         downloadProgress.nextSegment(segmentSize)
-
-        const result = await client.downloadToBuffer(
-          segmentStart,
-          segmentSize,
-          {
+        const abortTimeInMs =
+          options.abortTimeInMs === undefined ? 3600000 : options.abortTimeInMs
+        const result = await promiseWithTimeout(
+          abortTimeInMs,
+          client.downloadToBuffer(segmentStart, segmentSize, {
+            abortSignal: controller.signal,
```

Comment:
The doc seems to suggest that the async work will get aborted on timeout.
```
const signal = AbortController.timeout(1000);
doAsyncWork({ abortSignal: signal });
```

Do we still need to handle timeout from our side?

Reply:
Didn't come across this, let me give it a try! 👍🏽

Reply:
[This](https://www.npmjs.com/package/@azure/abort-controller) is doc I referred above.

Reply:
Added